swupd/delta.go

```diff
@@ -0,0 +1,160 @@
+package swupd
+
+import (
+	"fmt"
+	"os"
+	"sync"
+
+	"github.com/clearlinux/mixer-tools/helpers"
+	"github.com/pkg/errors"
+)
+
+// CreateDeltas creates all delta files between the previous and current
+// version of the supplied manifest, returning a list of files that
+// delta creation failed for.
+func CreateDeltas(manifest *Manifest, statedir string) ([]*File, error) {
```

Comment:
This function only creates deltas between a manifest's version and its previous version. What if we want to create deltas from an older version of that manifest, say 5 versions back?